src/Seabug/Contract/Buy.purs

```diff
@@ -45,7 +45,9 @@ mkBuyTxData nftData mScriptUtxos = do
     nftData
     mScriptUtxos
   let
-    nftCollection = unwrap (unwrap nftData).nftCollection
+    nftData' = unwrap nftData
+    nftCollection = unwrap nftData'.nftCollection
+    nftId = unwrap nftData'.nftId
```

Comment:
Given these, the nftPrice line below can probably be changed to just grab from `nftId`